### PR TITLE
Analyze sentiment with codegen

### LINKED ISSUE #13904: Codegen for AnalyzeSentiment

### PR DESCRIPTION
**Pull request by maririos**
Moves TA `AnalyzeSentiment` to auto-generated code.
It adds test to validate non-breaking changes

Considerations for breaking changes:
- Service sends Null for `Sentence.text`: 
  - TA 5.0.0 Text property is empty -> SDK was using [ToString](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.tostring?view=netcore-3.1) when parsing the JsonElement
  - With codegen `Text` is null -> Although the right way (using [GetString](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.getstring?view=netcore-3.1)), I decided to change the value to `String.Empty` to maintain consistency.

- Not supported `Document.Sentiment` value:
  - TA 5.0.0 -> System.ArgumentException
  - Codegen ->System.ArgumentOutOfRangeException => Because it derives from `ArgumentException`, it is [not](https://docs.microsoft.com/en-us/dotnet/core/compatibility/#exceptions) consider a breaking change.

- Not supported `Sentence.Sentiment` value:
  - TA 5.0.0 allows `Mixed` and after that it fails the same way as `Document.Sentiment`
  - codegen: Because of the serialization, it was throwing for anything that is not `Positive`, `Negative`, or `Neutral`, so I changed the type to `string` and I am doing the parsing.

Note: Try to keep the changes to a minimum, this change replaces existing behavior so it doesn't include support for Opinion Mining.

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/13904

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment_internal.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Azure.AI.TextAnalytics.Models
+{
+    [CodeGenModel("SentenceSentiment")]
+    internal partial struct SentenceSentiment_internal
+    {
+        /// <summary> The sentence text. </summary>
+        public string Text { get; }
+        /// <summary> The predicted Sentiment for the sentence. </summary>
+        public string Sentiment { get; }
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment_internal.cs:15]: Changed type from enum `SentenceSentimentValue` to `string`.
Reason => avoid breaking change so doing custom parsing

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 2]
```diff
@@ -1594,10 +1596,18 @@ public virtual Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinked
 
         internal virtual TextDocumentInput ConvertToDocumentInput(string document, string language, int id = 0)
             => new TextDocumentInput($"{id}", document) { Language = language ?? _options.DefaultLanguage };
-
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:1597]: This line removal was probably an accident. Don't let it block you, though.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 3]
```diff
@@ -1582,7 +1583,8 @@ public virtual Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinked
                 string id = item switch
                 {
                     TextDocumentInput tdi => tdi.Id,
-                    LanguageInput ldi => ldi.Id,
+                    LanguageInput li => li.Id,
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:1586]: Nice!

**Review by kinelski — APPROVED** [review]
LGTM!

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment_internal.cs [hunk 4]
```diff
@@ -5,12 +5,8 @@
 
 namespace Azure.AI.TextAnalytics.Models
 {
-    [CodeGenModel("DocumentSentimentValue")]
-    internal enum DocumentSentimentValue
+    [CodeGenModel("DocumentSentiment")]
+    internal partial struct DocumentSentiment_internal
```
Comment (tg-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment_internal.cs:9]: `_internal` isn't a very common naming convention for C# types.  I usually see just `Internal` like `DocumentSentimentInternal`.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs [hunk 5]
```diff
@@ -25,18 +27,26 @@ internal static TextAnalyticsError ConvertToError(TextAnalyticsError_internal er
             return new TextAnalyticsError(errorCode, message, target);
         }
 
-        internal static DetectedLanguage ConvertToDetectedLanguage(DocumentLanguage documentLanguage)
+        internal static List<TextAnalyticsWarning> ConvertToWarning(IReadOnlyList<TextAnalyticsWarning_internal> internalWarnings)
```
Comment (tg-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:30]: Nit - `ConvertToWarnings`?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment.cs [hunk 6]
```diff
@@ -17,6 +20,16 @@ internal SentenceSentiment(TextSentiment sentiment, string text, double positive
...
         }
 
+        internal SentenceSentiment(SentenceSentiment_internal sentenceSentiment)
+        {
+            // We shipped TA 5.0.0 Text == string.Empty if the service returned a null value for Text.
+            // Because we don't want to introduce a breaking change, we are transforming that null to string.Empty
+            Text = sentenceSentiment.Text ?? string.Empty;
+
+            ConfidenceScores = sentenceSentiment.ConfidenceScores;
+            Sentiment = (TextSentiment)Enum.Parse(typeof(TextSentiment), sentenceSentiment.Sentiment, ignoreCase: true);
```
Comment (tg-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment.cs:30]: Is this enum generated from the swagger so it'll always be able to parse the latest values?

**Review by tg-msft — APPROVED** [review]
The changes look good and I don't see anything that'll break.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment_internal.cs (hunk 4 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment_internal.cs:9]: interesting. I will rename this and I created https://github.com/Azure/azure-sdk-for-net/issues/14312 to do renames to other types in the library to not pollute this PR

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment.cs (hunk 6 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment.cs:30]: Yes. It is originally called DocumentSentimentValue. See https://github.com/Azure/azure-sdk-for-net/blob/22fdfa6f35b2ef2d4e1204509e3ffb803ee795de/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextSentiment.cs#L11

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs]: Added back

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs (hunk 3 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:1586]: ha! yous suggestion in a previous PR :)

**Pull request merged**